File: dayEight/python/TaskOne/test_task_one.py


from unittest import TestCase;
from task_one import *

class taskOneTest(TestCase):
    def test_ThatTheNumberIsEven(self) :
        number = 8
        
        self.assertTrue(isEven(number))

    def test_ThatTheNumberIsOdd(self) :
        number = 8
        
        self.assertTrue(isOdd(number))

    def test_ThatTheNumberIsOdd(self) :
        numbers = [5,4,3,2,1]
        
        expected = [6,8,4,4,2]
        actual = AddOneToEveryOddAndMultiplyEveryEvenIn(numbers)
        self.assertListEqual(actual, expected)

File: dayEight/java/ArraysOfIntegers/TestTaskOne.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestTaskOne {
    @Test
    public void testThatTheEvensInTheArrayAreMultipliedAndOddAreIncreasedByOneInTheGivenArray(){
    //Given
    int[] numbers = {2,8,14,5,1};
    
    int[] expected = TaskOne.AddOneToEveryOddAndMultiplyEveryEvenIn(numbers);
    int[] actual = {4,16,28,6,2};
    
    assertArrayEquals(expected, actual);
    }
    @Test
    public void testThatTheNumberIsEven(){
    int number = 6;
    
    assertTrue(TaskOne.isEven(number));
    }
    @Test
    public void testThatTheNumberIsOdd(){
    int number = 5;
    
    assertTrue(TaskOne.isOdd(number));
    }
}
